engine/runtime/src/main/java/org/enso/interpreter/runtime/data/hash/HashMapInsertAllNode.java

```diff
@@ -0,0 +1,62 @@
+package org.enso.interpreter.runtime.data.hash;
+
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.GenerateUncached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.nodes.NodeInfo;
+import org.enso.common.LanguageInfo;
+import org.enso.interpreter.node.expression.builtin.meta.EqualsNode;
+import org.enso.interpreter.node.expression.builtin.meta.HashCodeNode;
+
+@GenerateUncached
+@NodeInfo(
+    shortName = "MapInsertAll",
+    description = "Inserts all elements from the given container into a hash map",
+    language = LanguageInfo.ID)
+public abstract class HashMapInsertAllNode extends Node {
```

Comment:
Yeah, a node to put two `EnsoHashMap` together (even with a limit) is going to be needed!